hs_core/tasks.py

```diff
@@ -511,6 +512,60 @@ def replicate_resource_bag_to_user_zone_task(res_id, request_username):
         raise ValidationError("Resource {} does not exist in iRODS".format(res.short_id))
 
 
+@shared_task
+def delete_resource_task(resource_id, request_username=None):
+    """
+    Deletes a resource managed by HydroShare. The caller must be an owner of the resource or an
+    administrator to perform this function.
+    :param resource_id: The unique HydroShare identifier of the resource to be deleted
+    :return: resource_id if delete operation succeeds
+             raise an exception if there were errors.
+    """
+    res = utils.get_resource_by_shortkey(resource_id)
+    res_title = res.metadata.title
+    res_type = res.resource_type
+    resource_related_collections = [col for col in res.collections.all()]
```

Comment:
can you just use list() instead of the comphrehension?

Reply:
I just copied the old code over and put it in a task to avoid any potential regression issue. Do you mean do this ```resource_related_collections = list(res.collections.all())```? I'd prefer not to touch working code in this PR if possible to limit scope to moving the operation to async only.

Reply:
Ok, we can leave it